File: backend/src/main/kotlin/com/github/plplmax/planning/subjects/paired/PairedSubjects.kt
package com.github.plplmax.planning.subjects.paired

import com.github.plplmax.planning.groups.Group
import com.github.plplmax.planning.subjects.Subject
import kotlinx.serialization.Serializable

@Serializable
data class PairedSubjects(
    val id: Int,
    val firstSubject: Subject,
    val secondSubject: Subject,
    val groups: List<Group>,
    val count: Int
)

File: frontend/src/subjects/PairedSubjects.ts
import type { Group } from '@/groups/group'
import type { Subject } from './Subject'

export interface PairedSubjects {
  id: number
  firstSubject: Subject
  secondSubject: Subject
  count: number
  groups: Group[]
}

export interface PairedSubjectsUi {
  id: number
  firstSubject: Subject | null
  secondSubject: Subject | null
  count: number
  groups: Group[]
}
